[3.3.5] Missing NPC spells and quest credit

hiMalleus opened the pull request:
<!--- (**********************************)
      (** Fill in the following fields **)
      (**********************************) --->

**Changes proposed:**

-  Added chop spell to Venture Co. Straggler
- Added missing script for Enraged Spirits
- Fixed missing credit for Escaping from the Winterfin Caverns

**Target branch(es):** 3.3.5

- [x] 3.3.5

**Issues addressed:** 

Closes #20224 
Closes #20232 
Closes #20226 

**Tests performed:** 

Builds and works in game.



<!--- Notes
- Enable the setting "[√] Allow edits from maintainers." when creating your pull request.
- If this PR only contains SQL files, open a new issue instead and post or link the SQL in the issue.
- When adding new SQL files, name them 9999_99_99_99_db_name.sql to reduce the chance of possible merge conflicts.
--->

Commit:
Update zone_grizzly_hills.cpp

Commit:
Update zone_grizzly_hills.cpp

Commit:
Update zone_grizzly_hills.cpp

HannibalRoG commented on src/server/scripts/Northrend/zone_grizzly_hills.cpp:
```diff
@@ -612,6 +617,14 @@ class npc_venture_co_straggler : public CreatureScript
+		Chop_Timer = 8000;
+	    }
+	    else
+		Chop_Timer -= diff;
```
A lot of tab, dont use tab, change it for 4 spaces.

hiMalleus replied:
will remember it thanks

HannibalRoG reviewed (changes requested):
Lot of tab, change tab for 4 spaces.

Commit:
Update zone_grizzly_hills.cpp

Aokromes commented on src/server/scripts/Northrend/zone_grizzly_hills.cpp:
```diff
@@ -639,6 +652,7 @@ class npc_venture_co_straggler : public CreatureScript
         }
 };
 
+
```
remove :)

Commit:
Update zone_grizzly_hills.cpp

hiMalleus commented:
sorry for inconvenience on indentation!

joschiwald commented:
what does this pr change? i only see that you moved to eventmap handling to deprecated timers

sirikfoll commented:
Just revert all the changes and add something like this, please:
```diff
diff --git a/src/server/scripts/Northrend/zone_grizzly_hills.cpp b/src/server/scripts/Northrend/zone_grizzly_hills.cpp
index de646c0..cf752b1 100644
--- a/src/server/scripts/Northrend/zone_grizzly_hills.cpp
+++ b/src/server/scripts/Northrend/zone_grizzly_hills.cpp
@@ -573,6 +573,11 @@ public:
             me->SetReactState(REACT_AGGRESSIVE);
         }
 
+        void EnterCombat(Unit* /*who*/) override
+        {
+            _events.ScheduleEvent(EVENT_CHOP, Seconds(10), Seconds(12));
+        }
+
         void UpdateAI(uint32 diff) override
         {
             _events.Update(diff);
@@ -602,7 +607,7 @@ public:
                     case EVENT_CHOP:
                         if (UpdateVictim())
                             DoCastVictim(SPELL_CHOP);
-                        _events.ScheduleEvent(EVENT_CHOP, 10000, 12000);
+                        _events.Repeat(Seconds(10), Seconds(12));
                         break;
                     default:
                         break;

hiMalleus commented:
thanks for clearing it out sirikfoll, will do.

Commit:
Update zone_grizzly_hills.cpp

Aokromes commented on src/server/scripts/Northrend/zone_grizzly_hills.cpp:
```diff
@@ -599,19 +604,19 @@ class npc_venture_co_straggler : public CreatureScript
-                            DoCastVictim(SPELL_CHOP);
-                        _events.ScheduleEvent(EVENT_CHOP, 10000, 12000);
-                        break;
+		    case EVENT_CHOP:
```
tabs2spaces

hiMalleus replied:
I don't understand, if i tab it isn't in line with the rest.

hiMalleus replied:
Forgive my ignorance, i was doing changes directly through the website and not with gitextensions :$

Aokromes commented on src/server/scripts/Northrend/zone_grizzly_hills.cpp:
```diff
@@ -599,19 +604,19 @@ class npc_venture_co_straggler : public CreatureScript
             if (!UpdateVictim())
                 return;
-
+		
```
remove whiteline.

Commit:
Update zone_grizzly_hills.cpp

Commit:
Added spells to Enraged Spirits in SMV

Commit:
Update zone_shadowmoon_valley.cpp

Commit:
Indentation issues fixed

Commit:
Indentation issues

Commit:
More Indentation changes

hiMalleus commented:
Forgive a noob for his inexperience, i swear to god I learnt my lesson.

HannibalRoG commented on src/server/scripts/Outland/zone_shadowmoon_valley.cpp:
```diff
@@ -1390,7 +1390,22 @@ enum Enraged_Dpirits
-
+    
+    // ENRAGED WATER SPIRIT SPELLS
+    SPELL_STORMBOLT			                = 38032,
```
More tabs here...

HannibalRoG commented on src/server/scripts/Outland/zone_shadowmoon_valley.cpp:
```diff
@@ -1390,7 +1390,22 @@ enum Enraged_Dpirits
+    SPELL_ENRAGE                            = 8599,
+
+    // ENRAGED FIRE SPIRIT SPELLS - Will be using the enrage spell from Air Spirit
+    SPELL_FEL_FIREBALL			            = 36247,
```
and here...

HannibalRoG commented on src/server/scripts/Outland/zone_shadowmoon_valley.cpp:
```diff
@@ -1413,6 +1428,15 @@ enum Enraged_Dpirits
+    EVENT_ENRAGED_FIRE_SPIRIT			        = 2,
+    EVENT_ENRAGED_EARTH_SPIRIT			        = 3,
+    EVENT_ENRAGED_AIR_SPIRIT_CHAIN_LIGHTNING	= 4,
+    EVENT_ENRAGED_AIR_SPIRIT_HURRICANE		    = 5
```
and a lot more here...

HannibalRoG commented on src/server/scripts/Outland/zone_shadowmoon_valley.cpp:
```diff
@@ -1427,9 +1451,76 @@ class npc_enraged_spirit : public CreatureScript
+			default:
+				break;
+		}
+	}
```
and a tone here

HannibalRoG commented on src/server/scripts/Outland/zone_shadowmoon_valley.cpp:
```diff
@@ -1427,9 +1451,76 @@ class npc_enraged_spirit : public CreatureScript
+				DoCastSelf(SPELL_ENRAGE);
+
+		DoMeleeAttackIfReady();
+	}
```
same...

HannibalRoG commented on src/server/scripts/Outland/zone_shadowmoon_valley.cpp:
```diff
@@ -1488,6 +1580,9 @@ class npc_enraged_spirit : public CreatureScript
         }
+
+    private:
+	    EventMap _events;
```
one more here

HannibalRoG reviewed (changes requested):
You need to configure your editor so that when you press the tab key you will replace it with 4 spaces.

If you use notepad++ read https://stackoverflow.com/questions/455037/convert-tabs-to-spaces-in-notepad
If you use VS read https://stackoverflow.com/questions/14167033/visual-studio-replace-tab-with-4-spaces/14167067

Aokromes commented on src/server/scripts/Northrend/zone_grizzly_hills.cpp:
```diff
@@ -599,19 +604,18 @@ class npc_venture_co_straggler : public CreatureScript
-                            DoCastVictim(SPELL_CHOP);
-                        _events.ScheduleEvent(EVENT_CHOP, 10000, 12000);
-                        break;
+		            case EVENT_CHOP:
```
tabs2spaces

Commit:
Many more indentation changes

Aokromes commented on src/server/scripts/Northrend/zone_grizzly_hills.cpp:
```diff
@@ -600,18 +605,17 @@ class npc_venture_co_straggler : public CreatureScript
-                            DoCastVictim(SPELL_CHOP);
-                        _events.ScheduleEvent(EVENT_CHOP, 10000, 12000);
-                        break;
+			            if (UpdateVictim())
```
tabs2spaces.

Aokromes commented on src/server/scripts/Outland/zone_shadowmoon_valley.cpp:
```diff
@@ -1390,7 +1390,22 @@ enum Enraged_Dpirits
     NPC_ENRAGED_AIR_SPIRIT                  = 21060,
     NPC_ENRAGED_WATER_SPIRIT                = 21059,
-
+    
```
remove whitespaces

Commit:
tabs2spaces and whitespaces

Keader commented on src/server/scripts/Outland/zone_shadowmoon_valley.cpp:
```diff
@@ -1429,7 +1453,74 @@ class npc_enraged_spirit : public CreatureScript
+                }
+            }
+
+		if (!UpdateVictim())
```
it should be before of: https://github.com/TrinityCore/TrinityCore/pull/20225/files#diff-ba4f7087a55302e25b5ce01e7dcfdbbfR1479

Keader commented on src/server/scripts/Outland/zone_shadowmoon_valley.cpp:
```diff
@@ -1429,7 +1453,74 @@ class npc_enraged_spirit : public CreatureScript
+		if (!UpdateVictim())
+			return;
+
+		if (me->GetEntry() == NPC_ENRAGED_FIRE_SPIRIT || me->GetEntry() == NPC_ENRAGED_AIR_SPIRIT)
```
It should be in DamageTaken hook.

hiMalleus replied:
what do you mean DamageTaken hook, i only based that enrage logic from patchwerk it works fine in here

Keader replied:
This hook: https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/Northrend/Ulduar/Ulduar/boss_razorscale.cpp#L551
It's called when creature take some damage.
You can put this logic inside of this function

Keader commented on src/server/scripts/Outland/zone_shadowmoon_valley.cpp:
```diff
@@ -1429,7 +1453,74 @@ class npc_enraged_spirit : public CreatureScript
+			return;
+
+		if (me->GetEntry() == NPC_ENRAGED_FIRE_SPIRIT || me->GetEntry() == NPC_ENRAGED_AIR_SPIRIT)
+			if (HealthBelowPct(35) && !me->GetAura(SPELL_ENRAGE))
```
should be: !me->HasAura(SPELL_ENRAGE), you dont need get this aura...

hiMalleus replied:
If i don't get this aura then the npc is just going to be casting enrage all the time when below 35% health

hiMalleus replied:
I even tried using a boolean to validate the enrage but if the npc is channeling for example hurricane, then he turns it to true and doesn't cast the spell. That's the only reason i chose that logic.

Keader replied:
try: 
```cpp
if (HealthBelowPct(35) && !me->HasAura(SPELL_ENRAGE))
    DoCastSelf(SPELL_ENRAGE, true);
```

Keader commented on src/server/scripts/Outland/zone_shadowmoon_valley.cpp:
```diff
@@ -1429,7 +1453,74 @@ class npc_enraged_spirit : public CreatureScript
+            switch (me->GetEntry())
+            {
+                case NPC_ENRAGED_WATER_SPIRIT:
+                    _events.ScheduleEvent(EVENT_ENRAGED_WATER_SPIRIT, Seconds(0), Seconds(1));
```
You realy sure about it? 0~1 sec? sniffs show it?

hiMalleus replied:
I am not sure i don't have accurate data only guessed from fotage

Keader replied:
If you dont have access to retail (to sniff it)
ask someone on IRC if can do it.
We dont accept guessed values.

hiMalleus replied:
I closed it because you rather have this broken instead of using some temporary values. Looks like It is impossible to sniff things from retail right now because patch 7.3 has updated the exe to a powerful encryption or so i was told on #Trinity IRC.

Keader replied:
It's not a temp fix, it's hack.
Shauren should make a new sniffer soon, he always do. Or maybe Aokromes has some old sniff about this creatures. Is not only me... TC Team dont accept guessed values...

offl replied:
> We dont accept guessed values.

One half or more of scripts(SAI, CPP) has wrong guessed timers and scripting methods(how NPC will use ability, for example), so it's strange to hear.
Since TC replaced EAI by SAI some progress was made by ACID(EAI), but it never been ported to TC. Sometimes [you can find there](https://raw.githubusercontent.com/ACID-Scripts/WOTLK/master/acid_wotlk.sql) good scripts with timers from retail for some TBC locations(currently ACID has perfectly scripted Tanaris, for example, and some TBC heroic dungeons. While TC uses old ACID version).
Or you can made script based on other scrips.

Faq replied:
So this thing is only one what holds for back merge?

Aokromes replied:
old times where different, things changed with time, from when we don't had snifs to now.

Aokromes replied:
PS: even you don't have sniffer now, you can look at combat logs.

Faq replied:
I suggest to merge other code and leave this script hanging.

Keader reviewed (changes requested):
please change this stuffs

Commit:
Update victim before events update

Commit:
broggok death poison cloud

Added poison cloud on his death

Commit:
whitespaces

Shauren commented:
Honestly, for spell timing a retail combat log would work as well

Keader commented:
Looks a good option too.

hiMalleus commented:
I'll see if I can find someone to record me a log then

Commit:
Revert "broggok death poison cloud"

This reverts commit 7a9cf546ebcd3e2234e56ae36985dbde6df5a43b.

Commit:
Revert "whitespaces"

This reverts commit 3dc191c69f3a40be301a8dbcbad7af3836a64ad3.

Aokromes commented on src/server/scripts/Outland/HellfireCitadel/BloodFurnace/boss_broggok.cpp:
```diff
@@ -65,7 +65,7 @@ class boss_broggok : public CreatureScript
                 _EnterCombat();
                 Talk(SAY_AGGRO);
-            }
+            } 
```
remove tailing whitespace.

Commit:
Winterfin quest and wrong space

changed his state from Aggressive to Defensive as shown on video https://www.youtube.com/watch?v=apkZvnlkLX8 on minute 1:57, His cage door now opens for 20 seconds calculated from video and fixed a misstyped space on Broggok script

Commit:
Escape from the Winterfin Caverns

Fixed not getting credit for escorting the npc until the end.

hiMalleus commented:
**Enraged water spirit**

https://www.warcraftlogs.com/reports/Kcb34tDYw9pqVZjP/#type=casts&view=events&hostility=1&pins=0%24Separate%24%23244F4B%24damage%241%240.0.0.Any%2421059.0.0.NPC%24true%240.0.0.Any%24false%240&source=1&boss=0&difficulty=0

- Stormbolt cast every 17- 23 Seconds. The first one was cast approximately 1 second on entered combat. 

**Enraged Air Spirit**

https://www.warcraftlogs.com/reports/Kcb34tDYw9pqVZjP/#type=summary&view=events&hostility=1&pins=0%24Separate%24%23244F4B%24damage%241%240.0.0.Any%2421059.0.0.NPC%24true%240.0.0.Any%24false%240&source=6&boss=0&difficulty=0

- First Chain Lightning cast after 10 seconds entering combat and the following one 20 seconds after the first.

Commit:
Enraged Spirit timers and missing spell

Aokromes commented:
@Keader sooo this can be merged? the only doubtful thing is correct.

Aokromes committed:
Update zone_shadowmoon_valley.cpp
